src/libraries/Common/src/Interop/Linux/OpenLdap/Interop.Ldap.cs

```diff
@@ -67,6 +68,26 @@ internal static partial class Ldap
     {
         static Ldap()
         {
+            // Register callback that tries to load other libraries when the default library "libldap-2.4.so.2" not found
+            AssemblyLoadContext.Default.ResolvingUnmanagedDll += (assembly, ldapName) =>
+            {
+                IntPtr handle = IntPtr.Zero;
+
+                if (ldapName != Libraries.OpenLdap)
+                {
+                    return handle;
+                }
+
+                if (NativeLibrary.TryLoad("libldap-2.5.so.0", out handle) ||
+                    NativeLibrary.TryLoad("libldap-2.6.so.0", out handle) ||
```

Comment:
Open LDAP version 2.6 is in preview, not sure if the library name would be like this, need to check